ledger/src/blocktree.rs

```diff
@@ -273,61 +273,58 @@ impl Blocktree {
         }
     }
 
-    // Returns whether or not all iterators have reached their end
-    fn run_purge_batch(&self, from_slot: Slot, batch_end: Slot) -> Result<bool> {
-        let from_slot = Some(from_slot);
-        let batch_end = Some(batch_end);
-
+    fn run_purge_batch(&self, from_slot: Slot, to_slot: Slot) -> Result<bool> {
         let mut write_batch = self
             .db
             .batch()
             .expect("Database Error: Failed to get write batch");
-        let end = self
-            .meta_cf
-            .delete_slot(&mut write_batch, from_slot, batch_end)
-            .unwrap_or(false)
-            && self
-                .erasure_meta_cf
-                .delete_slot(&mut write_batch, from_slot, batch_end)
-                .unwrap_or(false)
-            && self
-                .data_shred_cf
-                .delete_slot(&mut write_batch, from_slot, batch_end)
-                .unwrap_or(false)
-            && self
-                .code_shred_cf
-                .delete_slot(&mut write_batch, from_slot, batch_end)
-                .unwrap_or(false)
-            && self
-                .transaction_status_cf
-                .delete_slot(&mut write_batch, from_slot, batch_end)
-                .unwrap_or(false)
-            && self
-                .orphans_cf
-                .delete_slot(&mut write_batch, from_slot, batch_end)
-                .unwrap_or(false)
-            && self
-                .index_cf
-                .delete_slot(&mut write_batch, from_slot, batch_end)
-                .unwrap_or(false)
-            && self
-                .dead_slots_cf
-                .delete_slot(&mut write_batch, from_slot, batch_end)
-                .unwrap_or(false)
-            && self
-                .db
-                .column::<cf::Root>()
-                .delete_slot(&mut write_batch, from_slot, batch_end)
-                .unwrap_or(false);
 
+        // delete range cf is not inclusive
+        let to_slot = to_slot.checked_add(1).unwrap_or_else(|| std::u64::MAX);
+        let columns_empty = self
+            .db
+            .delete_range_cf::<cf::SlotMeta>(&mut write_batch, from_slot, to_slot)
+            .unwrap_or_else(|_| false)
+            & self
```

Comment:
Also found a major bug. 

I forgot that `&&` short circuits the evaluation and so had to switch to `&` to make sure all columns were deleted properly.

This is probably why limit-ledger-size was not doing anything. :man_facepalming:

Reply:
Yay!

Reply:
Think you can easily backport part of this to 0.21 today?   We've had multiple slp1 validators run out of disk space already

Reply:
Early test results were not promising. There was no difference. I think with DeleteRange we need compaction to actually delete the slots. Otherwise it just marks a tombstone and calls it a day. I'll investigate today.

Reply:
Yeah i need to test if it fixes the older delete logic better than the DeleteRange pathway. I'll work on it today.